### PR TITLE
public api followup

### PR DESCRIPTION
**Pull request by tamird**
- public-api: ignore fixtures in git grep
- public-api: simplify and improve output

### CODE REVIEW

File: xtask/public-api/.gitattributes [hunk 1]
```diff
@@ -0,0 +1 @@
+/*.txt binary
```
Comment (dave-tucker) [inline xtask/public-api/.gitattributes:1]: I'm not a fan. This suppresses textual diffs when an API changes 😢

**Review by dave-tucker — CHANGES_REQUESTED** [review]
I'm not sure how this improves the output:

**BEFORE**
```console
Error: public api may have changed in one or more packages.
please bless by re-running this command with --bless
Errors:
public api for aya-bpf-bindings changed:
+pub const aya_sssssssbpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32
-pub const aya_bpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32

public api for aya-bpf-macros changed:
+pub proc macro aysssa_bpf_macros::#[xdp]
-pub proc macro aya_bpf_macros::#[xdp]
```

***AFTER***
```console
Error: aya-bpf-bindings public API changed; re-run with --bless. diff:
+pub const aya_sssssssbpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32
-pub const aya_bpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32

aya-bpf-macros public API changed; re-run with --bless. diff:
+pub proc macro aysssa_bpf_macros::#[xdp]
-pub proc macro aya_bpf_macros::#[xdp]
```

Things of note are:
- no newline after `diff`
- repetition of `re-run with --bless. diff:`

File: xtask/public-api/.gitattributes (hunk 1 above)
Comment (tamird) [reply xtask/public-api/.gitattributes:1]: It does, but you can still see it, I think. It's really upsetting when you're searching through code and the results are a lot of these .txt files.

Is there a way to get diffs while still ignoring these files in `git grep`?

**Comment by tamird** [comment]
> I'm not sure how this improves the output:
> 
> **BEFORE**
> 
> ```
> Error: public api may have changed in one or more packages.
> please bless by re-running this command with --bless
> Errors:
> public api for aya-bpf-bindings changed:
> +pub const aya_sssssssbpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32
> -pub const aya_bpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32
> 
> public api for aya-bpf-macros changed:
> +pub proc macro aysssa_bpf_macros::#[xdp]
> -pub proc macro aya_bpf_macros::#[xdp]
> ```
> 
> _**AFTER**_
> 
> ```
> Error: aya-bpf-bindings public API changed; re-run with --bless. diff:
> +pub const aya_sssssssbpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32
> -pub const aya_bpf_bindings::bindings::_bindgen_ty_32::BPF_WRITE_HDR_TCP_SYNACK_COOKIE: aya_bpf_bindings::bindings::_bindgen_ty_32::Type = 2u32
> 
> aya-bpf-macros public API changed; re-run with --bless. diff:
> +pub proc macro aysssa_bpf_macros::#[xdp]
> -pub proc macro aya_bpf_macros::#[xdp]
> ```
> 
> Things of note are:
> 
> * no newline after `diff`
> * repetition of `re-run with --bless. diff:`

The current output is bad when the package fails to build (or has other errors):

Before:
```
Error: public api may have changed in one or more packages.
please bless by re-running this command with --bless
Errors:
error checking public api for aya
Manifest must be for an actual package. `"Cargo.toml"` is a virtual manifest
public api for aya-obj changed:
-pub aya_obj::btf::BtfType::Foo

error checking public api for aya-log
Manifest must be for an actual package. `"Cargo.toml"` is a virtual manifest
```

After:
```
Error: public API errors:
aya failed to check public API: rustdoc_json::Builder::build
aya-obj public API changed; re-run with --bless. diff:
-pub aya_obj::btf::BtfType::Foo

aya-log failed to check public API: rustdoc_json::Builder::build
```

File: xtask/public-api/.gitattributes (hunk 1 above)
Comment (dave-tucker) [reply xtask/public-api/.gitattributes:1]: `git grep "NetlinkError" -- ':!*.txt'`

Comment (tamird) [reply xtask/public-api/.gitattributes:1]: Sorry, my question was too specific. I also want my editor to ignore these files.

Comment (dave-tucker) [reply xtask/public-api/.gitattributes:1]: If it's vscode (or similar) we can add to the workspace settings:
```json
  "search.exclude": {
    "xtask/public-api/*.txt": true
  }
  ```

Comment (tamird) [reply xtask/public-api/.gitattributes]: I guess that's the best we can do. Done.

**Review by dave-tucker — APPROVED** [review]

**Pull request merged**